spark/src/main/scala/geotrellis/spark/tiling/MapKeyTransform.scala

```diff
@@ -140,19 +141,18 @@ class MapKeyTransform(val extent: Extent, val layoutCols: Int, val layoutRows: I
      * array of  tuples which  is then  mapped-over to  produce an
      * array of KeyBounds.
      */
-    val tiles = new ConcurrentHashMap[(Int,Int), Unit]
+    val tiles: MSet[(Int, Int)] = MSet.empty
     val fn = new Callback {
       def apply(col: Int, row: Int): Unit = {
-        val tile: (Int, Int) = (bounds.colMin + col, bounds.rowMin + row)
-        tiles.put(tile, Unit)
+        tiles += ((bounds.colMin + col, bounds.rowMin + row))
       }
     }
 
     multiLine.lines.foreach { line => Rasterizer.foreachCellByLineStringDouble(line, rasterExtent)(fn) }
-    tiles.keys.asScala.map { case (col, row) => SpatialKey(col, row) }
+    tiles.map { case (col, row) => SpatialKey(col, row) }.toSet
```

Comment:
This is a drive by optimization, but it stands out given PR title: `SpatialKey` is just as good as `Tuple2` to be stored in the `MSet` and this map can be just avoided all together.

Reply:
My assumption is that @lossyrob had it as `Tuple2` because of some JVM optimizations that occur on a `Tuple2` of primitives. You have a point though, that said gain might be undone by the extra `.map`.